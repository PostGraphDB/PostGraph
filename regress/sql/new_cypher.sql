/*
 * Copyright (C) 2024 PostGraphDB
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */ 
-- Regression tests don't preload extensions, gotta load it first

LOAD 'postgraph';

SET search_path TO 'postgraph';
SET search_path TO DEFAULT;
SET search_path TO 'postgraph';

CREATE EXTENSION postgraph;
CREATE EXTENSION hstore CASCADE SCHEMA public;
CREATE EXTENSION IF NOT EXISTS hstore;
CREATE EXTENSION IF NOT EXISTS pg_trgm VERSION '1.3';

-- Basic Graph creation
CREATE GRAPH new_cypher;

-- Assign Graph to use
USE GRAPH new_cypher;

-- Reuse Name, should throw error
CREATE GRAPH new_cypher;

-- Graph Does not exist, should throw error
USE GRAPH new_cypher;

CREATE GRAPH new_cypher_2;
USE GRAPH new_cypher_2;
USE GRAPH new_cypher;

-- Old Grammar is at least partially plugged in.
RETURN 1 as a;

CREATE (n);

CREATE (n) RETURN n;

--CREATE (n) RETURN *;

MATCH (n) RETURN n;

MATCH (n) RETURN *;

CREATE TABLE tst (i text);
CREATE TABLE tst3 () INHERITS (tst);
CREATE TEMPORARY TABLE tst2 (i text);

CREATE TABLE tst4 (i text) PARTITION BY LIST (i);
CREATE TABLE tst5 (i text) PARTITION BY LIST (i COLLATE de_DE);
CREATE TABLE tst6 (i text) PARTITION BY LIST (i COLLATE de_DE text_ops);
CREATE TABLE tst7 (i text) USING heap;
CREATE TABLE tst8 (i text) WITH (fillfactor=70);

SELECT;
SELECT ALL;

SELECT *;
SELECT * FROM tst;
SELECT tst.i FROM tst;

SELECT a.i FROM tst AS a;
SELECT a.i FROM tst a;

SELECT a.i as j FROM tst a;


SELECT a.j FROM tst AS a(j);

SELECT i FROM tst WHERE i = i;
SELECT i FROM tst WHERE i > i;
--SELECT a.j FROM tst a(j);
SELECT a.* FROM tst as a;

SELECT a.* FROM tst as a ORDER BY a.i DESC;

SELECT a.i as j FROM tst a GROUP BY a.i;

SELECT a.i as j FROM tst a GROUP BY a.i HAVING a.i = a.i;

SELECT sum(salary) OVER w, avg(salary) OVER w
  FROM empsalary
  WINDOW w AS (PARTITION BY i ORDER BY i DESC);

SELECT a.i FROM tst a
UNION
SELECT a.i FROM tst a;

SELECT a.i FROM tst a
EXCEPT
SELECT a.i FROM tst a;

SELECT a.i FROM tst a
INTERSECT
SELECT a.i FROM tst a;

INSERT INTO tst SELECT 'Hello';


TABLE tst;
SELECT * FROM tst;

SELECT '"Hello"'::gtype = '"Hello"'::gtype;

SELECT '{"Hello": "World"}'::gtype <@ '"Hello"'::gtype;

SELECT 'Hello' LIKE 'Hello';

SELECT 'Hello' NOT LIKE 'Hello';

SELECT 'Hello' ILIKE 'hello';

SELECT 'Hello' NOT ILIKE 'hello';

SELECT 'Hello' SIMILAR TO 'Hello';

SELECT 'Hello' NOT SIMILAR TO 'Hello';

SELECT true AND false;
SELECT true OR false;
SELECT NOT false;

SELECT 1 IS NULL;
SELECT 1 IS NOT NULL;
SELECT 1 ISNULL;
SELECT 1 NOTNULL;

UPDATE tst SET i = 'World';
SELECT a.i FROM tst a;

DELETE FROM tst;
SELECT a.i FROM tst a;

SELECT a.i FROM ONLY tst a;

CREATE TABLE tbl1 (i int4) USING heap;
CREATE TABLE tbl2 (j int4) USING heap;

INSERT INTO tbl1 VALUES (1), (2);
INSERT INTO tbl2 VALUES (3), (2);

SELECT * FROM tbl1 AS t1 JOIN tbl2 AS t2 ON t1.i = t2.j;
SELECT * FROM tbl1 AS t1 LEFT JOIN tbl2 AS t2 ON t1.i = t2.j;
SELECT * FROM tbl1 AS t1 RIGHT JOIN tbl2 AS t2 ON t1.i = t2.j;
SELECT * FROM tbl1 AS t1 FULL JOIN tbl2 AS t2 ON t1.i = t2.j;

DROP GRAPH new_cypher CASCADE;
DROP GRAPH new_cypher_2 CASCADE;