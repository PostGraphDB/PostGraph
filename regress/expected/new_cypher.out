/*
 * Copyright (C) 2024 PostGraphDB
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */ 
-- Regression tests don't preload extensions, gotta load it first
LOAD 'postgraph';
SET search_path TO 'postgraph';
SET search_path TO DEFAULT;
SET search_path TO 'postgraph';
CREATE EXTENSION postgraph;
ERROR:  extension "postgraph" already exists
CREATE EXTENSION hstore CASCADE SCHEMA public;
CREATE EXTENSION IF NOT EXISTS hstore;
NOTICE:  extension "hstore" already exists, skipping
CREATE EXTENSION IF NOT EXISTS pg_trgm VERSION '1.3';
-- Basic Graph creation
CREATE GRAPH new_cypher;
NOTICE:  graph "new_cypher" has been created
 create_graph 
--------------
 
(1 row)

-- Assign Graph to use
USE GRAPH new_cypher;
 use_graph 
-----------
 
(1 row)

-- Reuse Name, should throw error
CREATE GRAPH new_cypher;
ERROR:  graph "new_cypher" already exists
-- Graph Does not exist, should throw error
USE GRAPH new_cypher;
 use_graph 
-----------
 
(1 row)

CREATE GRAPH new_cypher_2;
NOTICE:  graph "new_cypher_2" has been created
 create_graph 
--------------
 
(1 row)

USE GRAPH new_cypher_2;
 use_graph 
-----------
 
(1 row)

USE GRAPH new_cypher;
 use_graph 
-----------
 
(1 row)

-- Old Grammar is at least partially plugged in.
RETURN 1 as a;
 a 
---
 1
(1 row)

CREATE (n);
--
(0 rows)

CREATE (n) RETURN n;
                           n                            
--------------------------------------------------------
 {"id": 281474976710658, "label": "", "properties": {}}
(1 row)

--CREATE (n) RETURN *;
MATCH (n) RETURN n;
                           n                            
--------------------------------------------------------
 {"id": 281474976710657, "label": "", "properties": {}}
 {"id": 281474976710658, "label": "", "properties": {}}
(2 rows)

MATCH (n) RETURN *;
                           n                            
--------------------------------------------------------
 {"id": 281474976710657, "label": "", "properties": {}}
 {"id": 281474976710658, "label": "", "properties": {}}
(2 rows)

CREATE TABLE tst (i text);
CREATE TABLE tst3 () INHERITS (tst);
CREATE TEMPORARY TABLE tst2 (i text);
CREATE TABLE tst4 (i text) PARTITION BY LIST (i);
CREATE TABLE tst5 (i text) PARTITION BY LIST (i COLLATE de_DE);
CREATE TABLE tst6 (i text) PARTITION BY LIST (i COLLATE de_DE text_ops);
CREATE TABLE tst7 (i text) USING heap;
CREATE TABLE tst8 (i text) WITH (fillfactor=70);
SELECT;
--
(1 row)

SELECT ALL;
--
(1 row)

SELECT *;
ERROR:  SELECT * with no tables specified is not valid
LINE 1: SELECT *;
               ^
SELECT * FROM tst;
 i 
---
(0 rows)

SELECT tst.i FROM tst;
 i 
---
(0 rows)

SELECT a.i FROM tst AS a;
 i 
---
(0 rows)

SELECT a.i FROM tst a;
 i 
---
(0 rows)

SELECT a.i as j FROM tst a;
 j 
---
(0 rows)

SELECT a.j FROM tst AS a(j);
 j 
---
(0 rows)

SELECT i FROM tst WHERE i = i;
 i 
---
(0 rows)

SELECT i FROM tst WHERE i > i;
 i 
---
(0 rows)

--SELECT a.j FROM tst a(j);
SELECT a.* FROM tst as a;
 i 
---
(0 rows)

SELECT a.* FROM tst as a ORDER BY a.i DESC;
 i 
---
(0 rows)

SELECT a.i as j FROM tst a GROUP BY a.i;
 j 
---
(0 rows)

SELECT a.i as j FROM tst a GROUP BY a.i HAVING a.i = a.i;
 j 
---
(0 rows)

SELECT sum(salary) OVER w, avg(salary) OVER w
  FROM empsalary
  WINDOW w AS (PARTITION BY i ORDER BY i DESC);
ERROR:  syntax error at or near "("
LINE 1: SELECT sum(salary) OVER w, avg(salary) OVER w
                  ^
SELECT a.i FROM tst a
UNION
SELECT a.i FROM tst a;
 i 
---
(0 rows)

SELECT a.i FROM tst a
EXCEPT
SELECT a.i FROM tst a;
 i 
---
(0 rows)

SELECT a.i FROM tst a
INTERSECT
SELECT a.i FROM tst a;
 i 
---
(0 rows)

INSERT INTO tst SELECT 'Hello';
TABLE tst;
   i   
-------
 Hello
(1 row)

SELECT * FROM tst;
   i   
-------
 Hello
(1 row)

SELECT '"Hello"'::gtype = '"Hello"'::gtype;
 ?column? 
----------
 t
(1 row)

SELECT '{"Hello": "World"}'::gtype <@ '"Hello"'::gtype;
ERROR:  syntax error at or near "<@"
LINE 1: SELECT '{"Hello": "World"}'::gtype <@ '"Hello"'::gtype;
                                           ^
SELECT 'Hello' LIKE 'Hello';
 ?column? 
----------
 t
(1 row)

SELECT 'Hello' NOT LIKE 'Hello';
 ?column? 
----------
 f
(1 row)

SELECT 'Hello' ILIKE 'hello';
 ?column? 
----------
 t
(1 row)

SELECT 'Hello' NOT ILIKE 'hello';
 ?column? 
----------
 f
(1 row)

SELECT 'Hello' SIMILAR TO 'Hello';
 ?column? 
----------
 t
(1 row)

SELECT 'Hello' NOT SIMILAR TO 'Hello';
 ?column? 
----------
 f
(1 row)

SELECT true AND false;
 ?column? 
----------
 f
(1 row)

SELECT true OR false;
 ?column? 
----------
 t
(1 row)

SELECT NOT false;
 ?column? 
----------
 t
(1 row)

SELECT 1 IS NULL;
 ?column? 
----------
 f
(1 row)

SELECT 1 IS NOT NULL;
 ?column? 
----------
 t
(1 row)

SELECT 1 ISNULL;
 ?column? 
----------
 f
(1 row)

SELECT 1 NOTNULL;
 ?column? 
----------
 t
(1 row)

UPDATE tst SET i = 'World';
SELECT a.i FROM tst a;
   i   
-------
 World
(1 row)

DELETE FROM tst;
SELECT a.i FROM tst a;
 i 
---
(0 rows)

SELECT a.i FROM ONLY tst a;
 i 
---
(0 rows)

SELECT true is TRUE;
 ?column? 
----------
 t
(1 row)

SELECT true is not TRUE;
 ?column? 
----------
 f
(1 row)

SELECT true is false;
 ?column? 
----------
 f
(1 row)

SELECT true is not false;
 ?column? 
----------
 t
(1 row)

SELECT NULL is unknown;
 ?column? 
----------
 t
(1 row)

SELECT NULL is not unknown;
 ?column? 
----------
 f
(1 row)

SELECT 1 IS DISTINCT FROM 2;
 ?column? 
----------
 t
(1 row)

SELECT 1 IS NOT DISTINCT FROM 2;
 ?column? 
----------
 f
(1 row)

CREATE TABLE tbl1 (i int4) USING heap;
CREATE TABLE tbl2 (j int4) USING heap;
INSERT INTO tbl1 VALUES (1), (2);
INSERT INTO tbl2 VALUES (3), (2);
SELECT * FROM tbl1 AS t1 JOIN tbl2 AS t2 ON t1.i = t2.j;
 i | j 
---+---
 2 | 2
(1 row)

SELECT * FROM tbl1 AS t1 LEFT JOIN tbl2 AS t2 ON t1.i = t2.j;
 i | j 
---+---
 1 |  
 2 | 2
(2 rows)

SELECT * FROM tbl1 AS t1 RIGHT JOIN tbl2 AS t2 ON t1.i = t2.j;
 i | j 
---+---
 2 | 2
   | 3
(2 rows)

SELECT * FROM tbl1 AS t1 FULL JOIN tbl2 AS t2 ON t1.i = t2.j;
 i | j 
---+---
 1 |  
 2 | 2
   | 3
(3 rows)

SELECT * FROM tbl1 AS t1 WHERE t1.i IN (SELECT j FROM tbl2);
 i 
---
 2
(1 row)

SELECT * FROM tbl1 AS t1 WHERE t1.i NOT IN (SELECT j FROM tbl2);
 i 
---
 1
(1 row)

SELECT (SELECT * FROM tbl1 WHERE i = 1);
 i 
---
 1
(1 row)

SELECT * FROM tbl1 AS t1 WHERE EXISTS (SELECT j FROM tbl2 AS t2 WHERE t1.i = t2.j);
 i 
---
 2
(1 row)

SELECT ARRAY (SELECT * FROM tbl1);
 array 
-------
 {1,2}
(1 row)

SELECT 1 BETWEEN SYMMETRIC 0 AND 2;
 ?column? 
----------
 t
(1 row)

SELECT 1 BETWEEN 0 AND 2;
 ?column? 
----------
 t
(1 row)

SELECT 1 BETWEEN ASYMMETRIC 0 AND 2;
 ?column? 
----------
 t
(1 row)

DROP GRAPH new_cypher CASCADE;
NOTICE:  drop cascades to 2 other objects
DETAIL:  drop cascades to table new_cypher._ag_label_vertex
drop cascades to table new_cypher._ag_label_edge
NOTICE:  graph "new_cypher" has been dropped
 drop_graph 
------------
 
(1 row)

DROP GRAPH new_cypher_2 CASCADE;
NOTICE:  drop cascades to 2 other objects
DETAIL:  drop cascades to table new_cypher_2._ag_label_vertex
drop cascades to table new_cypher_2._ag_label_edge
NOTICE:  graph "new_cypher_2" has been dropped
 drop_graph 
------------
 
(1 row)

